how-to/synonym-expansion.md

```diff
@@ -0,0 +1,150 @@
+(We attention to need also register target synonym term to `synonyms`.)
+
+We set PGroonga's index to `term` for searching to fast and without uppercase and lowercase.
+
+We can register multiple synonyms into `synonyms` because type of `synonyms` is `text[]`.
+
+### Register synonyms
+
+We can register synonyms by inserting data into `synonyms`.
+
+For example, if we want to register "Window", "display", and "video display" as synonyms, we can insert these terms into the table that register synonyms as below.
+
+```sql
+INSERT INTO synonyms (term, synonyms) VALUES ('Window', ARRAY['Window', 'display', 'video display']);
+INSERT INTO synonyms (term, synonyms) VALUES ('display', ARRAY['display', 'Window', 'video display']);
+INSERT INTO synonyms (term, synonyms) VALUES ('video display', ARRAY['video display', 'Window', 'display']);
+```
+
+### Add synonyms
+
+There are three patterns to add synonyms.
+
+#### Add new synonyms
+
+We can add synonyms in the same way as register synonyms.
+For example, if we want to add "copy", "replicate", and "simulate" as synonyms, we insert these terms into the table that register synonyms as below.
+
+```sql
+INSERT INTO synonyms (term, synonyms) VALUES ('copy', ARRAY['copy', 'replicate', 'simulate']);
+INSERT INTO synonyms (term, synonyms) VALUES ('replicate', ARRAY['replicate', 'copy', 'simulate']);
+INSERT INTO synonyms (term, synonyms) VALUES ('simulate', ARRAY['simulate', 'copy', 'replicate']);
+```
+
+#### Add new synonyms in existing one
+
+If we want to add new synonyms in the existing one, we update the existing record.
+For example, if we also want to match "imitate" when we use "copy" as a search key word.
+
+```sql
+UPDATE synonyms SET synonyms = array_append(synonyms, 'imitate') WHERE term = 'copy';
+UPDATE synonyms SET synonyms = array_append(synonyms, 'imitate') WHERE term = 'replicate';
+UPDATE synonyms SET synonyms = array_append(synonyms, 'imitate') WHERE term = 'simulate';
+```
+
+"imitate" is new synonym. Therefore we add record into `synonyms` by using `INSERT` as below.
+
+```sql
+INSERT INTO synonyms (term, synonyms) VALUES ('imitate', ARRAY['imitate', 'copy', 'replicate', 'simulate']);
+```
+
+#### Modify exist synonyms
+
+If we want to modify an existing record, we update one.
+For example, if we want to modify "Windows" to "Window", we as below.
+
+```sql
+UPDATE synonyms SET synonyms = array_replace(synonyms, 'Windows', 'Window') WHERE term = 'display' OR term = 'video display' OR term = 'Windows';
+UPDATE synonyms SET term = 'Window' WHERE term = 'Windows';
+```
+
+#### Delete synonyms
+
+If we want to delete synonyms, we can delete record from `synonyms`.
+For example, if we want to delete "Window" from synonyms, we as below.
+
+```sql
+UPDATE synonyms SET synonyms = array_remove(synonyms, 'Window');
+DELETE synonyms WHERE term = 'Window';
+```
+
+### How to search of synonyms
+
+A search of synonyms uses [`pgroonga_query_expand` function][pgroonga_query_expand].
```

Comment:
**To** search of synonyms, we can use [`pgroonga_query_expand` function][pgroonga_query_expand]?

I think that this sentence's subject is _we_!